File: Template/Naked Objects Server/Template.Model/SubjectRepository.cs
using NakedObjects;
using System.Linq;

namespace Template.Model
{
    [Named("Subjects")]
    public class SubjectRepository
    {
        #region Injected Services
        //An implementation of this interface is injected automatically by the framework
        public IDomainObjectContainer Container { set; protected get; }
        #endregion

        public Subject CreateNewSubject() => Container.NewTransientInstance<Subject>();

        public IQueryable<Subject> AllSubjects() => Container.Instances<Subject>();

        public IQueryable<Subject> FindSubjectByName(string name) => AllSubjects().Where(c => c.Name.ToUpper().Contains(name.ToUpper()));
    }
}

File: Template/NOF2 Server/Model.Template/SubjectRepository.vb
Public Class SubjectRepository
    Public Property Container As IContainer

    Public Function CreateNewSubject() As Subject
        Return Container.CreateTransientInstance(Of Subject)()
    End Function

    Public Function AllSubjects() As IQueryable(Of Subject)
        Return Container.AllInstances(Of Subject)()
    End Function

    Public Function FindSubjectByName(ByVal name As String) As IQueryable(Of Subject)
        Return AllSubjects().Where(Function(c) c.Name.ToUpper().Contains(name.ToUpper()))
    End Function
End Class

